spring-pulsar/src/main/java/org/springframework/pulsar/annotation/ReactivePulsarListenerAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,785 @@
+ */
+public class ReactivePulsarListenerAnnotationBeanPostProcessor<V>
+		implements BeanPostProcessor, Ordered, ApplicationContextAware, InitializingBean, SmartInitializingSingleton {
+
+	private final LogAccessor logger = new LogAccessor(LogFactory.getLog(getClass()));
+
+	/**
+	 * The bean name of the default {@link ReactivePulsarListenerContainerFactory}.
+	 */
+	public static final String DEFAULT_REACTIVE_PULSAR_LISTENER_CONTAINER_FACTORY_BEAN_NAME = "reactivePulsarListenerContainerFactory";
+
+	private static final String THE_LEFT = "The [";
+
+	private static final String RESOLVED_TO_LEFT = "Resolved to [";
+
+	private static final String RIGHT_FOR_LEFT = "] for [";
+
+	private static final String GENERATED_ID_PREFIX = "org.springframework.Pulsar.ReactivePulsarListenerEndpointContainer#";
+
+	private ApplicationContext applicationContext;
+
+	private BeanFactory beanFactory;
+
+	private BeanExpressionResolver resolver;
+
+	private BeanExpressionContext expressionContext;
+
+	private ReactivePulsarListenerEndpointRegistry<?> endpointRegistry;
+
+	private String defaultContainerFactoryBeanName = DEFAULT_REACTIVE_PULSAR_LISTENER_CONTAINER_FACTORY_BEAN_NAME;
+
+	private final PulsarListenerEndpointRegistrar registrar = new PulsarListenerEndpointRegistrar(
+			ReactivePulsarListenerContainerFactory.class);
+
+	private final PulsarHandlerMethodFactoryAdapter messageHandlerMethodFactory = new PulsarHandlerMethodFactoryAdapter();
+
+	private Charset charset = StandardCharsets.UTF_8;
+
+	private final Set<Class<?>> nonAnnotatedClasses = Collections.newSetFromMap(new ConcurrentHashMap<>(64));
+
+	private final ListenerScope listenerScope = new ListenerScope();
+
+	private AnnotationEnhancer enhancer;
+
+	private final AtomicInteger counter = new AtomicInteger();
+
+	@Override
+	public int getOrder() {
+		return LOWEST_PRECEDENCE;
+	}
+
+	public void setEndpointRegistry(ReactivePulsarListenerEndpointRegistry<?> endpointRegistry) {
+		this.endpointRegistry = endpointRegistry;
+	}
+
+	public void setDefaultContainerFactoryBeanName(String containerFactoryBeanName) {
+		this.defaultContainerFactoryBeanName = containerFactoryBeanName;
+	}
+
+	public void setCharset(Charset charset) {
+		Assert.notNull(charset, "'charset' cannot be null");
+		this.charset = charset;
+	}
+
+	@Override
+	public void afterPropertiesSet() {
+		buildEnhancer();
+	}
+
+	private void buildEnhancer() {
+		if (this.applicationContext != null) {
+			List<AnnotationEnhancer> enhancers = this.applicationContext
+					.getBeanProvider(AnnotationEnhancer.class, false).orderedStream().toList();
+			if (!enhancers.isEmpty()) {
+				this.enhancer = (attrs, element) -> {
+					for (AnnotationEnhancer enh : enhancers) {
+						attrs = enh.apply(attrs, element);
+					}
+					return attrs;
+				};
+			}
+		}
+	}
+
+	@Override
+	public void afterSingletonsInstantiated() {
+		this.registrar.setBeanFactory(this.beanFactory);
+
+		this.beanFactory.getBeanProvider(PulsarListenerConfigurer.class)
```

Comment:
It gets the same bean as the imperative. Is it OK ?

Reply:
I think it is OK to do so, as we are not expecting or suggesting mixing the two paradigms in the same app.